src/series/ops/image.rs

```diff
@@ -1,19 +1,52 @@
-use crate::datatypes::{DataType, ImageType};
+use arrow2;
+use image;
+
+use crate::array::ops::image::DaftImageBuffer;
+use crate::datatypes::{logical::ImageArray, DataType, ImageType, NullArray};
 
-use crate::series::IntoSeries;
 use crate::{
     error::{DaftError, DaftResult},
-    series::Series,
+    series::{IntoSeries, Series},
 };
 
 impl Series {
     pub fn image_decode(&self) -> DaftResult<Series> {
         match self.data_type() {
             DataType::Binary => {
-                todo!("not implemented");
+                let arrow_array = self.to_arrow();
+                let arrow_array = arrow_array.as_any().downcast_ref::<arrow2::array::BinaryArray<i64>>().unwrap();
+                let mut img_bufs = Vec::<Option<DaftImageBuffer>>::with_capacity(arrow_array.len());
+                let mut cached_dtype: Option<DataType> = None;
+                // Load images from binary buffers.
+                // Confirm that all images have the same value dtype.
+                for row in arrow_array.iter() {
+                    let dyn_img = row.map(image::load_from_memory).transpose().map_err(|e| DaftError::ValueError(format!("Decoding image from bytes failed: {}", e)))?;
```

Comment:
would be cleaner if we implement the decoding logic in`DaftImageBuffer::decode(buf: &[u8])`.